### ISSUE TITLE
[Fonts API] Remove adding missing fonts into theme.json

### ISSUE THREAD

**Issue by hellofromtonya** [WordPress/gutenberg#51716]
Part of #41479.

Reference: https://github.com/WordPress/gutenberg/issues/41479#issuecomment-1597915077

## What problem does this address?

As shared in Ongoing Roadmap update https://github.com/WordPress/gutenberg/issues/41479#issuecomment-1597915077, the Fonts API will no longer inject / add fonts into theme.json. Rather, it will become read-only of the theme.json merged data.

## What is your proposed solution?

Remove the `WP_Fonts_Resolver::add_missing_fonts_to_theme_json()` and associated methods in the resolver.

Remove where `WP_Fonts_Resolver::add_missing_fonts_to_theme_json()` is invoked in `WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()`.

Remove all associated tests.

## Merge Caution - Wait for Font Library

Be careful of when to merge the PR that resolves this ticket. Removing this method should happen when the Font Library is already or is about to merged into `trunk`.

This method adds fonts that plugins register with the Fonts API. That functionality is needed for sites like WPCOM and any other site that is using a plugin to register fonts. Once the Font Library lands, then the PR for this ticket can be merged.

**Comment by hellofromtonya** [comment]
This issue is no longer valid as it's being resolved in the new [Font Face](https://github.com/WordPress/gutenberg/pull/51770/) and Font Library implementations.
